### PR TITLE
add Cody tutorial content to docs

### PR DESCRIPTION
**Pull request by cconcannon**

### CODE REVIEW

**Comment by MaedahBatool** [comment]
Hey @cconcannon came across this PR. Can you please move this content under `cody/use-cases` folder path? We already have one example for generating unit tests listed there. I think it will be a nice addition there. WDYT?

**Comment by cconcannon** [comment]
@MaedahBatool the content in this PR is applicable to the VS Code extension only. I've ported it directly from the Cody VS Code code base because it's useful content that belongs in the docs, and it's the framework that we use for the in-product tutorial, but for some reason it was never added to our docs. Since it's only applicable to the VS Code extension, do you still think it belongs in the cody/use-cases folder?

**Comment by cconcannon** [comment]
@MaedahBatool on second thought, I think that /cody/use-cases is a good place for the content

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Cody Tutorial
+
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:2]: <p className="subtitle">This tutorial gives you a quick walkthrough of leveraging Cody for autocompletion, chat, explaining code, and running commands in VS Code.</p>

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+## Code Autocomplete
+
+![](https://storage.googleapis.com/sourcegraph-assets/blog/vs-code-onboarding-walkthrough-dec-2023-cody-autocomplete-tsx.gif)
+
+Start writing code and Cody will complete the line (or the entire function) for you. Hit tab to accept the suggestion.
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:7]:
```suggestion
Start writing code and Cody will complete the line (or the entire function) for you. Click **Tab** to accept the suggestion.
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+**✨ Pro-tips for using Cody autocomplete**
+
+• Autocomplete uses the surrounding code and context to inform the suggestions, so if you need to guide it you can add a comment above the line you're editing.
+
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:12]:
```suggestion
```
Rmv extra space between bullets

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+• You can hover over the grey suggestion to see a toolbar of alternative suggestions, as well as other options such as accepting a single word at a time.
+
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:14]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+![](https://storage.googleapis.com/sourcegraph-assets/blog/blog-vscode-v018-release/blog-v018-context-controls-002.gif)
+
+Answer questions about general programming topics, or specific to your codebase, with Cody chat. You can choose between LLMs, @-mentions files and symbols, and enable enhanced repository-wide code context.
+
+You can start a chat at any time using the default keyboard shortcut of `Option` `/` on macOS and `Alt` `/` on Windows & Linux.
+
+**✨ Pro-tips for using Cody chat**
+
+• Enabling “Enhanced Context”, and tagging specific files and symbols, gives the selected LLM more context to help answer your questions more accurately.
+
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:28]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+• If you want to chat directly with the LLM without any additional context adding, disable "Enhanced Context".
+
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:30]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+• You can increase the quality of your chat responses by enabling "Search Context (Beta)" in your [Cody Settings](command:cody.status-bar.interacted).
+
+## Cody Commands
+
+![](https://storage.googleapis.com/sourcegraph-assets/blog/vs-code-onboarding-walkthrough-dec-2023-explain.gif)
+
+Cody has a range of commands for explaining code, generating unit tests, adding documentation, and more.
+
+To get started: select code in your editor, right click, and choose a command from the "Cody" menu.
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:39]:
```suggestion
To get started, select code in your editor, right-click, and choose a command from the "Cody" menu.
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+You can also use the [Cody: Commands Menu](command:cody.menu.commands) which has the default keyboard shortcut of `Option` `C` on macOS and `Alt` `C` on Windows & Linux.
+
+**✨ Pro-tips for using Cody commands**
+
+• You can build your own [Custom Commands (Beta)](https://sourcegraph.com/docs/cody/custom-commands) with custom prompts, output into chat or perform code edits, and more.
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:45]:
```suggestion
• You can build your own [Custom Commands (Beta)](/cody/capabilities/custom-commands) with custom prompts, output into chat or perform code edits, and more.
```
Add relative paths for links

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+## Edit Code
+
+You can ask Cody to perform code edits using the Edit Code command. This will edit code according to your instructions, giving you the chance to retry the edit with a new prompt, undo, or accept the change.
+
+To get started: select code in your editor, right click, and choose "Cody → Edit Code".
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:51]:
```suggestion
To get started, select code in your editor, right-click, and choose "Cody > Edit Code".
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+You can also use the default keyboard shortcut of `Option` `K` on macOS and `Alt` `K` on Windows & Linux.
+
+**✨ Pro-tips for code editing with Cody**
+
+• You can open the 💡 menu, with an "Cody: Edit Code" option, by selecting any line of code and using the keyboard short `Command` `.` on macOS or `Crtl` `.` on Windows & Linux.
+
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:58]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+• If you start an empty line and open the 💡 menu (using the keyboard short `Command` `.` on macOS or `Crtl` `.` on Windows & Linux) you can use "Cody: Generate Code" to generate new lines of code based on instructions.
+
+• You define your own custom code editing commands using [Custom Commands (Beta)](https://sourcegraph.com/docs/cody/custom-commands), by setting the [`commands.<id>.mode` property](https://sourcegraph.com/docs/cody/capabilities/commands#commands-id-mode).
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:61]:
```suggestion
• You define your own custom code editing commands using [Custom Commands (Beta)](/cody/capabilities/custom-commands), by setting the [`commands.<id>.mode` property](/cody/capabilities/commands).
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+## Explain Code
+
+<video autoPlay muted loop playsInline>
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:65]:
```suggestion
<video width="1920" height="1080" loop playsInline controls style={{ width: '100%', height: 'auto' }}>
 <source src="https://storage.googleapis.com/sourcegraph-assets/website/Product%20Animations/cody-explain-code-aug2023.mp4" type="video/mp4"/>
</video>
```

The correct format to add a video component.

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+    <source
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:66]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+        type="video/mp4"
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:67]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+        src="https://storage.googleapis.com/sourcegraph-assets/website/Product%20Animations/cody-explain-code-aug2023.mp4"
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:68]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+    />
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:69]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+</video>
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:70]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+
+Use Cody to get an in-depth explanation of any piece of code in any programming language.
+
+To get started: select code in your editor, right click, and choose "Cody → Explain Code".
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:74]:
```suggestion
To get started, select code in your editor, right-click, and choose "Cody > Explain Code".
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+
+You can also run this command from the [Cody: Commands Menu](command:cody.menu.commands), which has the default keyboard shortcut of `Option` `C` on macOS and `Alt` `C` on Windows & Linux.
+
+**✨ Pro-tips for understanding code with Cody**
+
+• Cody can also explain errors too, with the "Ask Cody to Explain" option in the 💡 menus, or by right clicking on any items in the "Problems" tab of VS Code.
+
+• You can define your own custom code explain commands to suit, such as prompt that requests an explanation focused on potential security issues, using [Custom Commands (Beta)](https://sourcegraph.com/docs/cody/custom-commands).
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:82]:
```suggestion
• You can define your own custom code to explain commands to suit, such as a prompt that requests an explanation focused on potential security issues, using [Custom Commands (Beta)](/cody/capabilities/custom-commands).
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+## Fix Code
+
+![](https://storage.googleapis.com/sourcegraph-assets/blog/vs-code-onboarding-walkthrough-dec-2023-ask-to-fix.gif)
+
+Something wrong with your code? Use Cody’s "Ask Cody to Fix" command to fix it, or explain the problem.
+
+**✨ Pro-tips for fixing code with Cody**
+
+• You can open the 💡 menu, with an "Ask Cody to Fix" option, by moving the cursor over any line of code with an error and using the keyboard short `Command` `.` on macOS or `Crtl` `.` on Windows & Linux.
+
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:93]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+Use Cody to get an in-depth explanation of any piece of code in any programming language.
+
+To get started: select code in your editor, right click, and choose "Cody → Explain Code".
+
+You can also run this command from the [Cody: Commands Menu](command:cody.menu.commands), which has the default keyboard shortcut of `Option` `C` on macOS and `Alt` `C` on Windows & Linux.
+
+**✨ Pro-tips for understanding code with Cody**
+
+• Cody can also explain errors too, with the "Ask Cody to Explain" option in the 💡 menus, or by right clicking on any items in the "Problems" tab of VS Code.
+
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:81]:
```suggestion
```

File: docs/cody/use-cases/vsc-tutorial.mdx [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+## Natural Language Search
+
+![](https://storage.googleapis.com/sourcegraph-assets/blog/vs-code-onboarding-walkthrough-dec-2023-natural-language.gif)
+
+Cody builds a Search index of your local files to make it easier to find what you’re looking for. Use a natural language query like “password hashing” or "connection retries" to quickly find and open the files that match your search.
+
+**✨ Pro-tips for using Cody’s Natural Language Search**
+
+• Increase the quality of your chat responses by enabling "Search Context (Beta)" in your [Cody Settings](command:cody.status-bar.interacted). This enables chat to use the natural language search index as a context source.
+
```
Comment (MaedahBatool) [inline docs/cody/use-cases/vsc-tutorial.mdx:125]:
```suggestion
```

**Review by MaedahBatool — CHANGES_REQUESTED** [review]
Leaving some comments. :)

File: docs/cody/use-cases/vsc-tutorial.mdx (hunk 2 above)
Comment (cconcannon) [reply docs/cody/use-cases/vsc-tutorial.mdx:7]: Tab isn't a button - it's the "Tab" key on the keyboard. I'll change to "Press **Tab**"

**Review by MaedahBatool — APPROVED** [review]

**Pull request merged**